scripts/eks-node-group-with-spot.yaml

```diff
@@ -0,0 +1,314 @@
+    Default: 0
+
+  NodeAutoScalingGroupMinSize:
+    Description: Minimum size of the Node Group.
+    Type: Number
+    Default: 0
+
+  NodeAutoScalingGroupMaxSize:
+    Description: Maximum size of the Node Group.
+    ConstraintDescription: |
+        If equal to minimum size, then cluster
+        autoscaler will be disabled for this group.
+    Type: Number
+    Default: 3
+
+  NodeVolumeSize:
+    Description: Node volume size (in GB)
+    Type: Number
+    Default: 20
+
+  OnDemandAllocationStrategy:
+    Description: AllocationStrategy for On-Demand instances
+    Type: String
+    Default: prioritized
+    AllowedValues:
+      - prioritized
+      - lowest-price
+
+  SpotAllocationStrategy:
+    Description: AllocationStrategy for Spot instances
+    Type: String
+    Default: lowest-price
+    AllowedValues:
+      - capacity-optimized
+      - lowest-price
+      - capacity-optimized-prioritized
+
+  OnDemandBaseCapacity:
+    Description: On-Demand instance base capacity
+    ConstraintDescription: |
+      The minimum amount of the Auto Scaling group's capacity
+      that must be fulfilled by On-Demand Instances.
+    Type: Number
+    Default: 0
+
+  OnDemandPercentageAboveBaseCapacity:
+    Description: Percentage of On-Demand instances
+    ConstraintDescription: |
+      Controls the percentages of On-Demand Instances and Spot Instances
+      for your additional capacity beyond OnDemandBaseCapacity (for example,
+      20 specifies 20% On-Demand Instances, 80% Spot Instances).
+    Type: Number
+    Default: 0
+
+  SpotInstancePools:
+    Description: Number of Spot instance pools
+    Type: Number
+    Default: 2
+
+  SpotMaxPrice:
+    Description: Maximum price paid for Spot instances
+    Type: String
+
+  ClusterName:
+    Description: The cluster name provided when the cluster was created.
+    ConstraintDescription: |
+      If it is incorrect, nodes will
+      not be able to join the cluster.
+    Type: String
+
+  NodeGroupName:
+    Description: Unique identifier for the Node Group.
+    Type: String
+
+  NodeInstanceProfile:
+    Description: IAM profile for node instances.
+    Type: String
+
+  NodeSecurityGroup:
+    Description: Security for the nodes in the cluster.
+    Type: String
+
+  Subnets:
+    Description: The subnets where workers can be created.
+    Type: List<AWS::EC2::Subnet::Id>
+
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+      - Label:
+          default: EKS Cluster
+        Parameters:
+          - ClusterName
+
+      - Label:
+          default: Group Configuration
+        Parameters:
+          - NodeGroupName
+          - NodeAutoScalingGroupMinSize
+          - NodeAutoScalingGroupMaxSize
+
+      - Label:
+          default: Instance Configuration
+        Parameters:
+          - NodeInstanceType
+          - NodeInstanceArch
+          - NodeGPUCount
+          - NodeInstanceProfile
+          - NodeImageId
+          - NodeVolumeSize
+
+    - Label:
```

Comment:
Just so we don't forget, this line is still misaligned!

Reply:
Thanks!